src/main/java/uk/nhs/hee/trainee/details/service/RabbitPublishService.java

```diff
@@ -72,6 +118,12 @@ public void publishCojSignedEvent(ProgrammeMembership programmeMembership) {
       event = new CojSignedEvent(firstPmId, conditionsOfJoining);
     }
 
-    rabbitTemplate.convertAndSend(rabbitExchange, routingKey, event);
+    outstandingConfirms.putIfAbsent(programmeMembership.getTisId(), programmeMembership);
+    try {
+      rabbitTemplate.convertAndSend(rabbitExchange, routingKey, event, correlationData);
+    } catch (AmqpException e) {
```

Comment:
If Rabbit dies, catching the exception here means the front-end is not left hanging